calico-cloud/threat/web-application-firewall.mdx

```diff
@@ -215,47 +204,102 @@ root@waf-attacker-1234:/ curl http://<host>//test/artists.php?artist=0+div+1+uni
+```
+2. Download bootstrap files.
+
+  Download the following required bootstrapping configuration files [Reference](https://medium.com/lightbaseio/web-application-firewall-in-go-feat-owasp-modsecurity-core-rule-set-3f97a26e3311) that are used by Tigera operator:
+    - `modsecdefault.conf`
+    - `crs-setup.conf`
+
+:::note
+
+The file names MUST be in all lowercase or they will fail to upload. They are loaded into ModSec before any REQUST-*.conf Core Rules Set files.
+
+:::
+
+```bash
+curl -O https://raw.githubusercontent.com/lsgroup/SmartReverseProxy/master/modsecdefault.conf
+curl https://raw.githubusercontent.com/coreruleset/coreruleset/v3.3/dev/crs-setup.conf.example > crs-setup.conf
+```
+
+3. Change your current directory to the my-ruleset folder where your core rules set files live. Create a configMap containing all the files downloaded into your new directory and replace the existing rule set with it:
+
+
+```bash
+kubectl create cm --dry-run=client --from-file=. -o yaml -n tigera-operator modsecurity-ruleset > ../my-ruleset.yaml
+kubectl replace -f ../my-ruleset.yaml
+```
+
+### Enable WAF using CLIs
+
+1. Create or update the [ApplicationLayer](https://docs.tigera.io/reference/installation/api#operator.tigera.io/v1.ApplicationLayer) resource to include the `webApplicationFirewall` field, and ensure it is set to `Enabled`.
 
 ```yaml
-apiVersion: projectcalico.org/v3
-kind: GlobalAlert
+apiVersion: operator.tigera.io/v1
+kind: ApplicationLayer
 metadata:
-  name: waf-new-alert-rule-info
+  name: tigera-secure
 spec:
-  summary: 'WAF new waf-alert-942100'
-  description: 'Test WAF Global Alert'
-  severity: 1
-  dataSet: waf
-  period: 1m
-  lookback: 1h
-  query: '"rule_info" IN {"*942100*"}'
-  threshold: 0
-  condition: gt
+  webApplicationFirewall: Enabled
 ```
+2. Annotate the services you want to enable WAF for.
 
-Apply the YAML to your cluster using: `kubectl apply -f test-demo-alert.yaml`
+  ```bash
+     kubectl annotate svc <service-name> -n <service-namespace> projectcalico.org/l7-logging=true
+  ```
 
-Now if a SQL Injection attack is detected for rule ID 942100, you will see the global alert in Manager UI, Activity, Alerts.
+   To remove annotation for the service.
 
-<img
-  src='/img/calico-enterprise/waf-alert.png'
-  alt='WAF alert'
-  width='600'
-/>
+  ```bash
+     kubectl annotate svc <service-name> -n <service-namespace> projectcalico.org/l7-logging-
+  ``
+
+#### Disable WAF for a service
+
+To disable WAF on a service, use the Actions menu on the WAF board, or use the following command:
+
+```
+kubectl annotate svc <service-name> -n <service-namespace> projectcalico.org/l7-logging-
+```
+
+#### Disable WAF feature
+
+To disable WAF, update the [ApplicationLayer](https://docs.tigera.io/reference/installation/api#operator.tigera.io/v1.ApplicationLayer) resource to include the `webApplicationFirewall` field, and ensure it is set to `Disabled`.
```

Comment:
Same as my other comment, seems like this shouldn't hard code `docs.tigera.io`. Also shouldn't this include `calico-cloud` in the path?

Reply:
yes, absolutely - we should never be putting hard coded references like that

Reply:
{{baseUrl}}